patches/tModLoader/Terraria.ModLoader.Core/ModCompile.cs

```diff
@@ -595,38 +595,14 @@ private class BuildingMod : LocalMod
 			}
 		}
 
-		private string DllRefPathForCompilation(BuildingMod mod, string dllName, bool xna)
-		{
-			string path = Path.Combine(mod.path, $"lib/{dllName}");
-			string refPath = DllRefPath(path, dllName, xna);
-			if (refPath != null)
-				return refPath;
-
-			throw new BuildException("Missing dll reference: " + path);
-		}
-
-		private IEnumerable<string> DllRefPathsForPackaging(BuildingMod mod, string dllName)
-		{
-			string path = Path.Combine(mod.path, $"lib/{dllName}");
-			string xnaRefPath = DllRefPath(path, dllName, true);
-			string fnaRefPath = DllRefPath(path, dllName, false);
-
-			string[] refPaths = new[] {xnaRefPath, fnaRefPath}.Where(p => p != null)
-				.Distinct().ToArray();
+		private string DllRefPath(BuildingMod mod, string dllName, bool? xna) {
+			var path = Path.Combine(mod.path, "lib", dllName + ".dll");
 
-			if (!refPaths.Any())
-				throw new BuildException("Missing dll reference: " + path);
-
-			return refPaths;
-		}
-
-		private string DllRefPath(string path, string dllName, bool xna)
-		{
-			//check if .XNA.dll /.FNA.dll exists when compiling for that platform
-			string suffixPlat = xna ? ".XNA.dll" : ".FNA.dll";
-			string pathPlat = path + suffixPlat;
-			if (File.Exists(pathPlat))
-				return pathPlat;
+			if (xna.HasValue) { //check for platform specific dll
```

Comment:
In this version of PackageMod, xna will always be null, so the check for platform-specific paths will never happen for noCompile mods. This functionality was the primary reason why I made this pull request and why I used SelectMany in PackageMod.